controllers/vmsingle_controller.go

```diff
@@ -73,6 +75,44 @@ func (r *VMSingleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 	if instance.Spec.ParsingError != "" {
 		return handleParsingError(instance.Spec.ParsingError, instance)
 	}
+	lastAppliedSingleSpec, err := instance.GetLastAppliedSpec()
+	if err != nil {
+		reqLogger.Error(err, "cannot parse last applied single spec")
+	}
+	singleChanges := deep.Equal(lastAppliedSingleSpec, &instance.Spec)
+	if len(singleChanges) == 0 {
+		// only update status by deployment pod status if single has no change
+		var currentDeploy appsv1.Deployment
+		err := r.Client.Get(ctx, types.NamespacedName{Name: instance.PrefixedName(), Namespace: instance.Namespace}, &currentDeploy)
+		if err != nil {
+			return result, fmt.Errorf("failed to get deployment for vmsingle %s: %w", req.NamespacedName, err)
+		}
+
+		instance.Status.ReadyReplicas = currentDeploy.Status.ReadyReplicas
```

Comment:
I think, we could skip Replicas fields changing, `VMSingle` always has only 1 replica.

Reply:
Oh, I stored ready pod replicas of deployment here to see if single is `operational`.

Reply:
Notice this note "if you need more - use vm cluster" until now :)
Fixed, not adding `ReadyReplicas` fields and keep rest replicas untouched for backward compatibility.